### DISCUSSION TITLE
Keeping seperate git repositories inside one another

### DISCUSSION THREAD

**Discussion by tom-burrows** [community/community#22468] [New to GitHub]
<p><strong>Context</strong><br>
I have a private git repository on GitHub which I use to store all of my University work. This has every module for every year all the way back to when I started.</p>
<p><strong>The Issue</strong><br>
However, one of my modules is a group project, for which we have to build and maintain a project within a GitLab repository.</p>
<p><strong>The Question</strong><br>
My question is, is there a way for me to store my /team-project repository within my /university repository?</p>
<p><strong>Limitations</strong><br>
Although the /university module is my own, the /team-project module has been created by a course admin / professor, so I can’t really do much beyond clone/push/pull.</p>
<p>When I try to clone it in, it says I can’t have a git, within a git, and offers to turn it into a submodule.</p>
<p>The only reason I want to do this is because I’m a bit of a neat-freak with regards to my directory trees and storage etc. So it doesn’t really need to be a part of the repository itself, I’d just like it to live within /university/year-2/team-project, regardless of whether it gets uploaded to and from the /university git.</p>
<p>Cheers!</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="TomBurrows101" data-post="1" data-topic="235002">
<div class="title">
<div class="quote-controls"></div>
 TomBurrows101:</div>
<blockquote>
<p>When I try to clone it in, it says I can’t have a git, within a git, and offers to turn it into a submodule.</p>
</blockquote>
</aside>
<p>A submodule is exactly what you need if you want to do this, see: <a href="https://git-scm.com/book/en/v2/Git-Tools-Submodules" class="inline-onebox">Git - Submodules</a></p>
<p>Note that in general keeping a lot of different projects in one repository is likely to cause trouble down the road. If it’s just a personal archive maybe not so much, but if it’s code you’ll want outside that it’s strongly recommended to use one repository per project (maybe class, paper, etc). For my graduation thesis I used one repository for the code I was developing, and another for my writing (LaTeX files, measurement output, graphs, etc.).</p>
